### ISSUE TITLE
PhysFS_init now returns 0 in 5.2.5

### ISSUE THREAD

**Issue by Giacom** [liballeg/allegro5#1029]
Hi, I just recently upgraded my Allegro project from 5.2.4 to 5.2.5.

I cannot get PhysFS to initialise. Before it worked just fine but now it will return 0, the error code for unsuccessfully initialising, without including any error code in getLastError or getLastErrorCode.

I'm developing on Windows 10 64-bit and downloaded Allegro through nuget.

I was able to workaround this problem by replacing my physfs.lib and physfs.h with the version from 5.2.4.

**Comment by SiegeLord** [comment]
So the difference between the two Nuget packages is that we upgraded PhysFS from 2.0.3 to 3.0.1. 

It looks like this is the issue described here: https://icculus.org/pipermail/physfs/2018-October/001274.html.

What I'll do is patch in the fix here (https://hg.icculus.org/icculus/physfs/rev/ece6769c0676) and push out a new version of the deps + allegro.

Thanks for reporting this!

**Comment by dos1** [comment]
```diff
@@ -575,7 +574,7 @@
          *  psize. Also note that the second parameter can't be
          *  NULL or the function fails.
          */
-        rc = pGetDir(accessToken, &dummy, &psize);
+        rc = pGetDir(accessToken, NULL, &psize);
```

Love that comment :D

**Comment by fatcerberus** [comment]
I'm very confused by the patch - the comment (as @dos1 points out) says the second parameter can't be `NULL` and yet the fix was to pass `NULL` for that very parameter?  What?!

**Comment by dos1** [comment]
It just seems to be pretty confusing piece of code. The commit message mentions API change in Windows 10, while the code itself actually seems to want this function to fail.

It's funny nevertheless ;)

**Comment by rcgordon** [comment]
Yeah, it's a mess. The NULL comment is due to Windows 95 requirements, which aren't a going concern anymore. I'm going to try to push a PhysicsFS 3.0.2 shortly to resolve this.

**Comment by SiegeLord** [comment]
Ah, awesome. I'll just wait for the 3.0.2 and upgrade to that.

**Comment by rcgordon** [comment]
> The NULL comment is due to Windows 95 requirements

Actually, it appears to be something that affects at least WinXP, but only if you build with an older Visual Studio...it's a mystery. I released PhysicsFS 3.0.2 today with something that works with both win32 behaviors: https://icculus.org/pipermail/physfs/2019-March/001280.html

**Comment by SiegeLord** [comment]
Alright, I've uploaded package version 5.2.5.1 with PhysFS 3.0.2. Things should work now.
